### PR TITLE
Update docs and tests for LLVMFuzzerTestOneInput

### PR DESCRIPTION
**Pull request by benrimmington**
* Add note on libFuzzer availability to docs.
* Use parameters suitable for UnsafeRawBufferPointer.
  - (Swift uses `Int` for `size_t`.)
* Replace `#if os(...)` checks with `canImport` checks.
* Move slowest test from [test/Fuzzing][] to [validation-test/Sanitizers][].
* Remove superfluous test file.

[test/Fuzzing]: <https://github.com/apple/swift/tree/4d000f02b31acff0aee4aa489d402341d4cd9212/test/Fuzzing>
[validation-test/Sanitizers]: <https://github.com/apple/swift/tree/master/validation-test/Sanitizers>

### CODE REVIEW

File: docs/libFuzzerIntegration.md [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
----------------------
-
-Swift compiler comes with a built-in `libFuzzer` integration.
-In order to use it on a file `myfile.swift`, we define an entry point fuzzing function
-with a `@_cdecl("LLVMFuzzerTestOneInput")` annotation:
+# libFuzzer Integration
 
+Custom builds of the Swift toolchain (including development snapshots)
+have a built-in `libFuzzer` integration. In order to use it on a file
+`myfile.swift`, we define an entry point fuzzing function with a
```
Comment (harlanhaskins) [inline docs/libFuzzerIntegration.md:5]: While you're updating these, any chance you can reword this to avoid `we`?

File: docs/libFuzzerIntegration.md [hunk 2]
```diff
@@ -1,37 +1,39 @@
...
+`@_cdecl("LLVMFuzzerTestOneInput")` annotation:
 
 ```swift
-@_cdecl("LLVMFuzzerTestOneInput") public func fuzzMe(Data: UnsafePointer<CChar>, Size: CInt) -> CInt{
-    // Test our code using provided Data.
-  }
+@_cdecl("LLVMFuzzerTestOneInput")
+public func test(_ start: UnsafeRawPointer, _ count: Int) -> CInt {
+  let bytes = UnsafeRawBufferPointer(start: start, count: count)
+  // TODO: Test our code using the provided bytes.
```
Comment (harlanhaskins) [inline docs/libFuzzerIntegration.md:12]: Nitpick: `the code`, instead of `our code` now that we're avoiding `we` 😄

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: test/Fuzzing/fuzzer_test.swift [hunk 3]
```diff
@@ -8,21 +8,21 @@
...
-@_cdecl("LLVMFuzzerTestOneInput") public func fuzzOneInput(Data: UnsafePointer<CChar>, Size: CLong) -> CInt {
-  if (Size >= 3 && Data[0] == 65 && Data[1] == 66 && Data[2] == 67) {
-    fputs("Crash!", stdout);
-    fflush(stdout);
-    exit(1);
+@_cdecl("LLVMFuzzerTestOneInput")
+public func test(_ start: UnsafeRawPointer, _ count: Int) -> CInt {
+  let bytes = UnsafeRawBufferPointer(start: start, count: count)
+  if bytes.starts(with: "ABC".utf8) {
+    fputs("Crash!", stdout)
```
Comment (harlanhaskins) [inline test/Fuzzing/fuzzer_test.swift:23]: I suppose this could just be `print` now...

Comment (benrimmington) [reply test/Fuzzing/fuzzer_test.swift:23]: I did try `print` earlier, and it works, but the `fflush` is still required. I thought [`exit`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/exit.html) is supposed to "flush all open streams"?

I also tried `fatalError`, and redirecting stderr to FileCheck, so that Darwin/Glibc/MSVCRT isn't required. But I couldn't get that to work.

**Comment by benrimmington** [comment]
All checks have passed.

Should the fuzzer test be moved from [test/Fuzzing][] to [test/Sanitizers][]?

Or should it be moved to [validation-test/Sanitizers][], because it is the slowest test on swift-ci?

> `454.80s: Swift(macosx-x86_64) :: Fuzzing/fuzzer_test.swift`

[test/Fuzzing]:               <https://github.com/apple/swift/tree/master/test/Fuzzing>
[test/Sanitizers]:            <https://github.com/apple/swift/tree/master/test/Sanitizers>
[validation-test/Sanitizers]: <https://github.com/apple/swift/tree/master/validation-test/Sanitizers>

**Comment by harlanhaskins** [comment]
Yeah that should definitely be moved to validation-test at the very least.

**Review by harlanhaskins — APPROVED** [review]

**Comment by harlanhaskins** [comment]
@swift-ci smoke test

**Comment by benrimmington** [comment]
All checks have passed.

@harlanhaskins Thanks for your help.

**Comment by harlanhaskins** [comment]
Thanks, @benrimmington!
